File: app/src/main/java/com/monk/rxjava2/imetate1/Subscriber.java
package com.monk.rxjava2.imetate1;

/**
 * @author monk
 * @date 2019-02-26
 */
public abstract class Subscriber<T> implements Observer<T>, Subsription {
    private volatile boolean isSubscribe;

    @Override
    public void unSubsribe() {
        isSubscribe = true;
    }

    @Override
    public boolean isUnSubscribe() {
        return isSubscribe;
    }
}

File: module_fragment/src/main/java/com/monk/modulefragment/rxjava2/imetate1/Subscriber.kt
package com.monk.modulefragment.rxjava2.imetate1

/**
 * @author monk
 * @date 2019-02-26
 */
abstract class Subscriber<T> : Observer<T>, Subsription {
    @Volatile
    private var isSubscribe = false
    override fun unSubsribe() {
        isSubscribe = true
    }

    override fun isUnSubscribe(): Boolean {
        return isSubscribe
    }
}